dissect/target/plugins/os/windows/lnk.py

```diff
@@ -154,7 +163,7 @@ def lnk(self, path: Optional[str]) -> Iterator[LnkRecord]:
                     target_ctime=target_ctime,
                 )
 
-    def lnk_entries(self, path: Optional[str] = None) -> Iterator[TargetPath]:
+    def lnk_entries(self, path: Optional[str] = None) -> Iterator[Path]:
```

Comment:
```suggestion
    def lnk_entries(self, path: Optional[str] = None) -> Iterator[TargetPath]:
```

One small thing, the return type should remain `TargetPath`. Input typing can be broader if the function can deal with any generic `Path` (sub)class, but return types should be specific so you know what (sub)class to exepect.